.github/workflows/python-package.yaml

```diff
@@ -53,20 +53,20 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy3.8', 'pypy3.9']
-        networkx-version: ['2.4','2.5', '2.6', '2.7', '2.8', '3.0']
+        python-version: ['3.9', '3.10', '3.11', '3.12', 'pypy3.9', 'pypy3.10']
+        networkx-version: ['2.4','2.5', '2.6', '2.7', '2.8', '3.0', '3.1', '3.2']
         include:
-          - python-version: '3.7'
-            networkx-version: '2.6'
-          - python-version: '3.7'
-            networkx-version: '1.11'
           - python-version: '3.8'
             networkx-version: '1.11'
+          - python-version: '3.8'
+            networkx-version: '2.4'
+          - python-version: '3.8'
+            networkx-version: '3.1' # as of 3.2, python 3.9 is not supported by networkx
         exclude:
             # missing wheels for scipy at these combos
-          - python-version: 'pypy3.9'
+          - python-version: 'pypy3.10'
             networkx-version: '2.6'
-          - python-version: 'pypy3.8'
+          - python-version: 'pypy3.9'
             networkx-version: '2.6'
```

Comment:
the top-level `python-version` and `networkx-version` lists create a matrix, and the `include` adds additional single runs with specific variables, while the `exclude` removes single runs that are specified in the matrix. so, the jobs that eventually get created are basically a list combining the matrix + includes - excludes

its similar to the gitlab matrix spec but not as flexible since you can only do one matrix